ext/oci8/oci8_statement.c

```diff
@@ -1066,20 +1074,15 @@ int php_oci_bind_post_exec(zval *data)
 					/* int curr_element_length = strlen(((text *)bind->array.elements)+i*bind->array.max_length); */
 					int curr_element_length = bind->array.element_lengths[i];
 					if ((i < (int) bind->array.old_length) && (entry = zend_hash_get_current_data(hash)) != NULL) {
-						zval_dtor(entry);
+						zval_ptr_dtor(entry);
 						ZVAL_STRINGL(entry, (char *)(((text *)bind->array.elements)+i*bind->array.max_length), curr_element_length);
 						zend_hash_move_forward(hash);
 					} else {
-						add_next_index_stringl(bind->zval, (char *)(((text *)bind->array.elements)+i*bind->array.max_length), curr_element_length);
+						add_next_index_stringl(zv, (char *)(((text *)bind->array.elements)+i*bind->array.max_length), curr_element_length);
 					}
 				}
 				break;
 		}
-	} else if ((Z_TYPE_P(bind->zval) == IS_TRUE) || (Z_TYPE_P(bind->zval) == IS_FALSE)) {
```

Comment:
in PHP-7 FALSE and TRUE are encoded by TYPE. The LVAL() of FALSE and TRUE is undefined.
ZVAL_BOOL(bind->zval, ...); just sets type to IS_FALSE or IS_TRUE.
So that code changed type IS_FALSE/IS_TRUE into IS_FALSE/IS_TRUE, depended on a uninitialised value.

May be you mind Z_TYPE_P(bind->zval) == IS_LONG ? (to perform LONG -> FALSE/TRUE conversion)
Then this chunk should be restored and fixed.

Reply:
This function php_oci_bind_post_exec() is to convert the data returned from Oracle database to PHP variables. In case of boolean data, we bind the buffer of zval (which is the field of zend_value in zval by using &Z_LVAL_P()) with Oracle and Oracle would put the returned value (1 or 0) in this buffer . Therefore, at this point we look at the value in the buffer to determine whether bind->zval should be changed to IS_FALSE or IS_TRUE. So it's actually not depended on an uninitialised value, but a value returned from database.

 If you think it's OK, we can restore this piece of code and change "bind->zval" to "zv".

Reply:
This function php_oci_bind_post_exec() is to convert the data returned from Oracle database to PHP variables. In case of boolean data, we bind a buffer of zval (which is the field of zend_value in zval by using &Z_LVAL_P()) with Oracle and Oracle would put the returned value (1 or 0) in this buffer. Therefore, at this point we look at the value in the buffer to determine whether bind->zval should be changed to IS_FALSE or IS_TRUE. So it's actually not depended on an uninitialised value, but a value returned from database.

 If you think it's ok, we can restore this piece of code and change "bind->zval" to "zv".

Reply:
OK. Then this needs to be restored.